src/main/java/duke/command/CommandParser.java

```diff
@@ -0,0 +1,65 @@
+package duke.command;
+
+import duke.commons.exceptions.DukeException;
+
+public class CommandParser {
+
+    public static CommandType parseCommand(String userInput) throws DukeException {
+        String[] splitInput = userInput.split("\\s+", 2);
+        String command = splitInput[0];
+        try {
+            return CommandType.valueOf(command.toUpperCase());
+        } catch (IllegalArgumentException e) {
+            throw new DukeException("Invalid Command: " + command);
+        }
+    }
+
+    public static int parseTaskIndex(String userInput) throws DukeException {
+        String[] splitInput = userInput.split("\\s+", 2);
+        try {
+            int idx = Integer.parseInt(splitInput[1]) - 1;
+            return idx;
+        } catch (ArrayIndexOutOfBoundsException e) {
+            throw new DukeException("Please provide a task index");
+        } catch (IllegalArgumentException e) {
+            throw new DukeException("Invalid task index: " + splitInput[1]);
+        }
+    }
+
+    public static String parseToDo(String userInput) throws DukeException {
```

Comment:
Maybe it will be good to create new clases for the parsing functions? (I know u will in the future) ❤️